srml/system/src/lib.rs

```diff
@@ -753,6 +755,42 @@ impl<T: Trait> Module<T> {
 	}
 }
 
+/// Weight limit check and increment.
+#[derive(Encode, Decode, Clone, Eq, PartialEq)]
+pub struct CheckWeight<T: Trait + Send + Sync>(PhantomData<T>);
+
+impl<T: Trait + Send + Sync> SignedExtension for CheckWeight<T> {
+	type AccountId = T::AccountId;
+	// TODO TODO: make sure that this has to be here and not in validate.
```

Comment:
it should be in both; `validate` happens on the way from the txq into the authored block, `pre_dispatch` is when verifying an (already authored) block.

Reply:
will add it once the operational stuff is merged in.